ql/src/java/org/apache/hadoop/hive/ql/exec/Utilities.java

```diff
@@ -4272,6 +4273,21 @@ public static void addTableSchemaToConf(Configuration conf,
       LOG.info("schema.evolution.columns and schema.evolution.columns.types not available");
     }
   }
+  public static void setPartitionColumnsToConf(Configuration conf, TableScanOperator tableScanOp) {
+    TableScanDesc scanDesc = tableScanOp.getConf();
+    if (scanDesc != null && scanDesc.getTableMetadata() != null) {
+      List<String> partitionColsList = scanDesc.getTableMetadata().getPartColNames();
+      if (!partitionColsList.isEmpty()) {
+        conf.set(IOConstants.PARTITION_COLUMNS, String.join(",", partitionColsList));
+      }
+    } else {
+      LOG.info(IOConstants.PARTITION_COLUMNS + " not available");
```

Comment:
Is this else condition placed correctly ?  Seems like the INFO message  for partition columns not available  should be logged when partitionColsList.isEmpty() returns true.

Reply:
Thanks for catching this. I will move it to the correct place.